src/ngraph/op/gather.cpp

```diff
@@ -208,3 +211,80 @@ shared_ptr<Node> op::v1::Gather::clone_with_new_inputs(const OutputVector& new_a
     check_new_args_count(this, new_args);
     return make_shared<v1::Gather>(new_args.at(PARAMS), new_args.at(INDICES), new_args.at(AXIS));
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    inline bool evaluate(const HostTensorPtr& arg0,
+                         const HostTensorPtr& arg1,
+                         const HostTensorPtr& out,
+                         size_t axis)
+    {
+        using T = typename element_type_traits<ET>::value_type;
```

Comment:
Somewhere around here you need to set the output shape/element type based on the input argument so that the dynamic input case works.

Reply:
added, also a unit test